backend/src/main/java/rocks/inspectit/gepard/agentmanager/configuration/service/GitService.java

```diff
@@ -0,0 +1,74 @@
+/* (C) 2024 */
+package rocks.inspectit.gepard.agentmanager.configuration.service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import rocks.inspectit.gepard.agentmanager.exception.GitOperationException;
+
+@Slf4j
+@Service
+public class GitService {
+
+  private static final String FILE = "configuration.json";
+
+  @Value("${inspectit-config-server.configurations.local-path}")
+  private String localRepoPath;
+
+  private Git git;
+
+  public void commit() {
+    try {
+      git.add().addFilepattern(FILE).call();
+      git.commit().setMessage("update configuration").call();
+    } catch (GitAPIException e) {
+      throw new GitOperationException("Failed to commit changes to local repository", e);
+    }
+  }
+
+  public void updateFileContent(String content) {
+    try {
+      File file = new File(localRepoPath + "/configuration.json");
+      java.nio.file.Files.write(file.toPath(), content.getBytes());
```

Comment:
Same thread-safety problem with writing into a file.

(Even tough we currently don't really have multiple threads writing the config, it would be nice to add some protection mechanism for writing together with reading)